shotover/src/transforms/opensearch/mod.rs

```diff
@@ -0,0 +1,119 @@
+use crate::tcp;
+use crate::transforms::{
+    Messages, Transform, TransformBuilder, TransformConfig, Transforms, Wrapper,
+};
+use crate::{
+    codec::{opensearch::OpenSearchCodecBuilder, CodecBuilder, Direction},
+    transforms::util::{
+        cluster_connection_pool::{spawn_read_write_tasks, Connection},
+        Request,
+    },
+};
+use anyhow::{anyhow, Result};
+use async_trait::async_trait;
+use serde::Deserialize;
+use std::time::Duration;
+use tokio::sync::oneshot;
+use tracing::trace;
+
+#[derive(Deserialize, Debug)]
+pub struct OpenSearchSinkSingleConfig {
+    #[serde(rename = "remote_address")]
+    address: String,
+    connect_timeout_ms: u64,
+}
+
+#[typetag::deserialize(name = "OpenSearchSink")]
```

Comment:
This is the name used by deserialization.
This needs to be OpenSearchSinkSingle.